utils/python/CIME/case_test.py

```diff
@@ -6,7 +6,14 @@
 from CIME.utils import expect, find_system_test, append_status
 from CIME.SystemTests.system_tests_common import *
 
-import sys
+import sys, signal
+
+def _signal_handler(*_):
+    expect(False, "Problem timed out!")
```

Comment:
Rather than 'Problem timed out', I'm thinking this should be more general, like, "Job killed due to signal" - because other things can send signals like the user deleting the job, right?